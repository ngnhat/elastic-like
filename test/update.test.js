/**
 * Created by ngnhat on Sat May 25 2019
 */
const Store = require('../index');

describe('update a document:', () => {
  it('non exists', () => {
    const store = new Store();
    store.update({ Id: 1, Code: 'aaa', Name: 'quick brown fox' });

    expect(store.search('fox')).toMatchSnapshot();
  });

  it('exists', () => {
    const store = new Store();

    store.add({ Id: 1, Code: 'aaa', Name: 'quick brown fox' });
    store.update({ Id: 1, Code: 'abc', Name: 'quick fox updated.' });

    expect(store.search('aaa')).toEqual([]);
    expect(store.search('brown')).toEqual([]);
    expect(store.search('updated')).toMatchSnapshot();
  });
});
