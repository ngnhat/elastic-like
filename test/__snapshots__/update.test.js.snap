// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update a document: exists 1`] = `
Array [
  Object {
    "score": 0.28768207245178085,
    "source": Object {
      "Code": "abc",
      "Id": 1,
      "Name": "quick fox updated.",
    },
  },
]
`;

exports[`update a document: non exists 1`] = `
Array [
  Object {
    "score": 0.28768207245178085,
    "source": Object {
      "Code": "aaa",
      "Id": 1,
      "Name": "quick brown fox",
    },
  },
]
`;
